cmd/file/io.go

```diff
@@ -0,0 +1,235 @@
+	scheme := "http"
+	if ssl {
+		scheme += "s"
+	}
+	u := &url.URL{
+		Host:   fmt.Sprintf("%s:%d", host, port),
+		Scheme: scheme,
+		Path:   "bzz-chunk",
+	}
+	return &ApiStore{
+		Client:  http.DefaultClient,
+		baseUrl: u.String(),
+	}
+}
+
+// Put implements storage.Putter.
+func (a *ApiStore) Put(ctx context.Context, mode storage.ModePut, chs ...swarm.Chunk) (exist []bool, err error) {
+	for _, ch := range chs {
+		addr := ch.Address().String()
+		buf := bytes.NewReader(ch.Data())
+		url := strings.Join([]string{a.baseUrl, addr}, "/")
+		res, err := a.Client.Post(url, "application/octet-stream", buf)
+		if err != nil {
+			return nil, err
+		}
+		if res.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("upload failed: %v", res.Status)
+		}
+	}
+	return nil, nil
+}
+
+// Get implements storage.Getter.
+func (a *ApiStore) Get(ctx context.Context, mode storage.ModeGet, address swarm.Address) (ch swarm.Chunk, err error) {
+	addressHex := address.String()
+	url := strings.Join([]string{a.baseUrl, addressHex}, "/")
+	res, err := http.DefaultClient.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("chunk %s not found", addressHex)
+	}
+	chunkData, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+	ch = swarm.NewChunk(address, chunkData)
+	return ch, nil
+}
+
+// LimitWriteCloser limits the output from the application.
+type LimitWriteCloser struct {
+	io.WriteCloser
+	total int64
+	limit int64
+}
+
+// NewLimitWriteCloser creates a new LimitWriteCloser.
+func NewLimitWriteCloser(w io.WriteCloser, c int64) io.WriteCloser {
+	return &LimitWriteCloser{
+		WriteCloser: w,
+		limit:       c,
+	}
+}
+
+// Write implements io.Writer.
+func (l *LimitWriteCloser) Write(b []byte) (int, error) {
+	if l.total+int64(len(b)) > l.limit {
+		return 0, errors.New("overflow")
+	}
+	c, err := l.WriteCloser.Write(b)
+	l.total += int64(c)
+	return c, err
+}
+
+// Join reads all output from the provided joiner.
+func JoinReadAll(j file.Joiner, addr swarm.Address, outFile io.Writer) error {
+	r, l, err := j.Join(context.Background(), addr)
+	if err != nil {
+		return err
+	}
+	// join, rinse, repeat until done
+	data := make([]byte, swarm.ChunkSize)
+	var total int64
+	for i := int64(0); i < l; i += swarm.ChunkSize {
+		cr, err := r.Read(data)
+		if err != nil {
+			return err
+		}
+		total += int64(cr)
+		cw, err := outFile.Write(data[:cr])
+		if err != nil {
+			return err
+		}
+		if cw != cr {
+			return fmt.Errorf("short wrote %d of %d for chunk %d", cw, cr, i)
+		}
+	}
+	if total != l {
+		return fmt.Errorf("received only %d of %d total bytes", total, l)
+	}
+	return err
+}
+
+func SetLogger(cmd *cobra.Command, verbosityString string) (logger logging.Logger, err error) {
+	v := strings.ToLower(verbosityString)
+	switch v {
+	case "0", "silent":
+		logger = logging.New(ioutil.Discard, 0)
+	case "1", "error":
+		logger = logging.New(cmd.OutOrStderr(), logrus.ErrorLevel)
```

Comment:
not sure about stderr as the default for everything in this block

Reply:
It would be a very strange thing to log to stdout at all, from a cli usage perspective.